### ISSUE TITLE
About the gumble softmax

### ISSUE THREAD

**Issue by YoungBigBird1985** [yoonholee/MT-net#6]
gumbel_hard = tf.cast(tf.equal(gumbel_softmax, tf.reduce_max(gumbel_softmax, 1, keep_dims=True)), tf.float32)
mask = tf.stop_gradient(gumbel_hard - gumbel_softmax) + gumbel_softmax

For the above code, it seems to choose the maximal index marked with "1" and all other indexes set to 0?, it means you only allow 1 element go through the gradient descent, which is too strict...?
